app/scripts/controllers/swaps.js

```diff
@@ -187,15 +183,24 @@ export default class SwapsController {
         this.setSwapsErrorKey(QUOTES_NOT_AVAILABLE_ERROR)
```

Comment:
In this branch, execution would continue through down to line 203 and it would throw an exception.

Would it make sense to return here instead? Are any of the subsequent lines useful if `newQuotes` is empty? Is this really something that can happen, and if so in what circumstances?

Reply:
Maybe not such a good idea, see here: https://github.com/MetaMask/metamask-extension/pull/9536#issuecomment-706246280